pkg/kubelet/winstats/winstats.go

```diff
@@ -0,0 +1,288 @@
+
+	// root (node) container
+	infos["/"] = *c.createRootContainerInfo()
+
+	containers, err := c.dockerClient.ContainerList(context.Background(), dockertypes.ContainerListOptions{})
+
+	if err != nil {
+		return nil, err
+	}
+
+	for _, container := range containers {
+		containerInfo, err := c.createContainerInfo(&container)
+
+		if err != nil {
+			return nil, err
+		}
+
+		infos[container.ID] = *containerInfo
+	}
+
+	return infos, nil
+}
+
+func (c *Client) WinMachineInfo() (*cadvisorapi.MachineInfo, error) {
+	hostname, err := os.Hostname()
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &cadvisorapi.MachineInfo{
+		NumCores:       runtime.NumCPU(),
+		MemoryCapacity: c.memoryPhysicalCapacityBytes,
+		MachineID:      hostname,
+	}, nil
+}
+
+func (c *Client) WinVersionInfo() (*cadvisorapi.VersionInfo, error) {
+	dockerServerVersion, err := c.dockerClient.ServerVersion(context.Background())
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &cadvisorapi.VersionInfo{
+		KernelVersion:    dockerServerVersion.KernelVersion,
+		DockerVersion:    dockerServerVersion.Version,
+		DockerAPIVersion: dockerServerVersion.APIVersion,
+	}, nil
+}
+
+func (c *Client) createRootContainerInfo() *cadvisorapiv2.ContainerInfo {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	var stats []*cadvisorapiv2.ContainerStats
+
+	stats = append(stats, &cadvisorapiv2.ContainerStats{
+		Timestamp: time.Now(),
+		Cpu: &cadvisorapi.CpuStats{
+			Usage: cadvisorapi.CpuUsage{
+				Total: c.cpuUsageCoreNanoSeconds,
+			},
+		},
+		Memory: &cadvisorapi.MemoryStats{
+			WorkingSet: c.memoryPrivWorkingSetBytes,
+			Usage:      c.memoryCommitedBytes,
+		},
+	})
+
+	rootInfo := cadvisorapiv2.ContainerInfo{
+		Spec: cadvisorapiv2.ContainerSpec{
+			HasCpu:    true,
+			HasMemory: true,
+			Memory: cadvisorapiv2.MemorySpec{
+				Limit: c.memoryPhysicalCapacityBytes,
+			},
+		},
+		Stats: stats,
+	}
+
+	return &rootInfo
+}
+func (c *Client) createContainerInfo(container *dockertypes.Container) (*cadvisorapiv2.ContainerInfo, error) {
+
+	spec := cadvisorapiv2.ContainerSpec{
+		CreationTime:     time.Unix(container.Created, 0),
+		Aliases:          []string{},
+		Namespace:        "docker",
+		Labels:           container.Labels,
+		Envs:             map[string]string{},
+		HasCpu:           true,
+		Cpu:              cadvisorapiv2.CpuSpec{},
+		HasMemory:        true,
+		Memory:           cadvisorapiv2.MemorySpec{},
+		HasCustomMetrics: false,
+		CustomMetrics:    []cadvisorapi.MetricSpec{},
+		HasNetwork:       true,
+		HasFilesystem:    false,
+		HasDiskIo:        false,
+		Image:            container.Image,
+	}
+
+	var stats []*cadvisorapiv2.ContainerStats
+	containerStats, err := c.createContainerStats(container)
+
+	if err != nil {
+		return nil, err
+	}
+
+	stats = append(stats, containerStats)
+	return &cadvisorapiv2.ContainerInfo{Spec: spec, Stats: stats}, nil
+}
+
+func (c *Client) createContainerStats(container *dockertypes.Container) (*cadvisorapiv2.ContainerStats, error) {
```

Comment:
It doesn't seem this function is using Windows specific API. Is Linux using different method for container?